### ISSUE TITLE
Undim screen whenever it's close to a turn

### ISSUE THREAD

**Issue by CaldeiraG** [organicmaps/organicmaps#6045]
**Is your feature request related to a problem? Please describe.**

I have set OM to "allow the screen to dim" while also trying to conserve battery on my phone.

This means that if you're close to a turn it will not show you the upcoming turn unless you tap on the screen.

**Describe the solution you would like**

Whenever you have an incoming instruction from OM, allow the screen to undim the screen to show the map.

The distance before you need to undim the screen needs to be different across all navigation modes, so while a driver needs to know in advance for quite a distance (maybe 1500m), a cyclist only needs instruction 500m before

**Describe alternatives you have considered**

I haven't tried other OSM navigation apps but Waze has this feature, I don't have any footage at the moment of the effect

**Comment by denilsonsa** [comment]
From my experience, this is the Waze behavior:

* You have to manually go to Settings, Battery Saver, and enable it only for the current drive.
* The next time you drive (or the next directions, I don't know), the option will be reset and you will have to remember to set it again. Most likely, it's so cumbersome to enable that you won't even bother.
* While driving at around 80km/h, it seems it brightened up the screen if there was any turn in the next 1.4km. I don't know if this threshold changes with the speed.
* Randomly, it decided to keep the screen bright.
* I believe it disables the battery saving mode if the phone is charging.

In my opinion, the Waze behavior is less than ideal. Or just plain bad.

In my opinion, a proper battery saving mode should be:

* Toggle once at settings, and gets remembered forever.
* It is engaged regardless of the battery charging percentage.
    * Some apps decide to be helpful by enabling battery saving only when the battery is low. I think that's stupid, because if the battery saving was enabled all the time, my battery wouldn't be so low so quickly.
* It is engaged regardless of charging state.
    * Because navigation uses a lot of energy (GPS, compass, bright screen, TTS, re-routing), having any feature that will reduce the usage is welcome.
    * If the energy usage is lowered while the phone is being charged, it will likely be charged a little bit faster.
    * If the user is cycling, very likely the phone is charging from a power bank, so energy savings are still very important.
* Different behaviors for different navigation modes
    * While driving, the user needs guidance further before a turn.
    * While cycling, the user needs guidance closer to a turn.
    * While walking… Well, the user probably has free hands and should be able to turn off and on the display at will. But maybe the user is skating instead of walking.

What should be the behavior? I don't know for sure, but I have a few ideas (brainstorm):

* Minimum saving: dim the screen.
* Medium saving: render a minimum set of instructions on top of a black screen. This can theoretically yield some savings on OLED screens due to fewer pixels being lit. Not rendering the map can also save some CPU/GPU usage.
* Maximum saving: completely turn off the screen.

**Comment by CaldeiraG** [comment]
> You have to manually go to Settings, Battery Saver, and enable it only for the current drive.

It will actually prompt you to enable Battery Saver (like traffic/hazards alerts) once you're below 20%

> The next time you drive (or the next directions, I don't know), the option will be reset and you will have to remember to set it again. Most likely, it's so cumbersome to enable that you won't even bother.
> While driving at around 80km/h, it seems it brightened up the screen if there was any turn in the next 1.4km. I don't know if this threshold changes with the speed.
> Randomly, it decided to keep the screen bright.

Based on my experience, the screen will turn on depending on your speed (my screen turned once at 2km before the turn at about 110kmh, but further testing may be useful)

The screen will stay on for a bit longer than those 2km but eventually dims out if your next direction is relatively far away

> I believe it disables the battery saving mode if the phone is charging.

I can't recall it but it seems to be the case.

Overall I don't think the Waze implemention is bad but I do agree on your points on keeping the battery saver mode always-on while also having the suggestion to turn it on on the current drive.

**Comment by denilsonsa** [comment]
> It will actually prompt you to enable Battery Saver […] once you're below 20%

Yeah, I hate when apps do that. If the battery is already low, it's too late to start saving battery, as the battery will be dead before my destination. The battery saving should have started since the beginning. 😒
